src/openms/source/ANALYSIS/ID/SiriusAdapterAlgorithm.cpp

```diff
@@ -183,7 +184,31 @@ namespace OpenMS
     {
       return tmp_ms_file_;
     }
-    
+
+    std::vector<String> SiriusAdapterAlgorithm::sortSiriusWorkspacePathsByScanIndex(const std::vector<String>& siriusworkspacepaths)
+    {
+      std::vector<String> sorted_siriusworkspacepaths;
+      std::vector<SiriusAdapterAlgorithm::SiriusWorkspaceIndex> indices;
+
+      for (size_t i = 0; i < siriusworkspacepaths.size(); i++)
+      {
+        SiriusAdapterAlgorithm::SiriusWorkspaceIndex wsi;
+        wsi.array_index = i;
+        wsi.scan_index = SiriusMzTabWriter::extractScanIndex(siriusworkspacepaths[i]);
+        indices.emplace_back(wsi);
```

Comment:
if you add a constructor as chris suggested you could use in-place construction with: indices.emplace_back({i,SiriusMzTabWriter::extractScanIndex(siriusworkspacepaths[i])})